hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -686,9 +686,24 @@ def _validate_deadline(x):
                 "deadline=%r is invalid, because it is impossible to meet a "
                 "deadline <= 0.  Use deadline=None to disable deadlines." % (x,)
             )
+        # check if x is larger than timedelta's max value (1000000000 days - 1 microsecond)
+        if x >= 86400000000000000:
+            raise InvalidArgument(
+                "deadline=%r is invalid, because it is impossible to meet a "
+                "deadline > datetime.timedelta.max. Use deadline=None to disable deadlines." % (x,)
+            )
+        if x is None:
```

Comment:
Let's split this out to an earlier clause before checking whether `x` is a number - if it's `None`, we don't need to convert it or check the bounds.

(this is actually why the tests are currently failing too - comparing None to the max timedelta seconds is an error)